src/angular-app/languageforge/lexicon/shared/share-with-others/user-management.component.ts

```diff
@@ -58,6 +59,18 @@ export class UserManagementController implements angular.IController {
     });
   }
 
+  onOwnershipTransfer($event: {target: Partial<User>}) {
+    var newOwnerId = $event.target.id;
+    this.projectService.transferOwnership(newOwnerId).then(() => {
+      this.loadMemberData();
+      this.project.ownerRef.id = newOwnerId;
+    });
+  }
+
+  currentUserIsOwnerOrAdmin(){
+    return this.project.ownerRef.id === this.session.data.userId || this.session.hasSiteRight(this.sessionService.domain.USERS, this.sessionService.operation.VIEW);
```

Comment:
Since `this.session.hasSiteRight(this.sessionService.domain.USERS, this.sessionService.operation.VIEW)` is a bit cryptic, how about adding a new function to `session.service.ts` for this:
```
isSystemAdmin() {
  return this.hasSiteRight(this.sessionService.domain.USERS, this.sessionService.operation.VIEW);
}
```

Reply:
I like this suggestion, and think we could make role-checking easier for other roles as well through methods like this. But what do you think @megahirt @longrunningprocess ? The "hasSiteRight" method with the domain and operation params seems to be the way role-checking occurs throughout the project right now

Reply:
How did you know that `hasSiteRight(USERS, VIEW)` was the right way to check this?

Reply:
I'm finding issues on my other PR with permissions checking so I will revisit this as well

Reply:
I won't go into the history of why we have permissions distinct from roles (the need for that in our current system is debatable and probably unnecessary complexity) but essentially the permission sets (domain + operation) are hard-coded in an [initialization function in PHP called SystemRoles.php](https://github.com/sillsdev/web-languageforge/blob/chore/removeWebsiteClass/src/Api/Model/Shared/Rights/SystemRoles.php#L27).  If you understand this function (which is regrettably somewhat hard to read), then you can know that admins have a site right of User+View.